src/java.base/share/classes/java/lang/StringCoding.java

```diff
@@ -522,85 +48,18 @@ final String requestedCharsetName() {
-        resultCached = new ThreadLocal<>() {
-            protected SoftReference<Result> initialValue() {
-                return new SoftReference<>(new Result());
-            }};
-    private static Result resultCached() {
-        SoftReference<Result> sr = resultCached.get();
-        Result r;
-        if (sr == null || (r = sr.get()) == null) {
-            r = new Result();
-            resultCached.set(new SoftReference<>(r));
-        }
-        return r;
-    }
-
-    ////////////////////////// ascii //////////////////////////////
-
-    private static Result decodeASCII(byte[] ba, int off, int len) {
-        Result result = resultCached();
-        if (COMPACT_STRINGS && !hasNegatives(ba, off, len)) {
-            return result.with(Arrays.copyOfRange(ba, off, off + len),
-                               LATIN1);
-        }
-        byte[] dst = new byte[len<<1];
-        int dp = 0;
-        while (dp < len) {
-            int b = ba[off++];
-            putChar(dst, dp++, (b >= 0) ? (char)b : repl);
-        }
-        return result.with(dst, UTF16);
-    }
-
-    private static byte[] encodeASCII(byte coder, byte[] val) {
-        if (coder == LATIN1) {
-            byte[] dst = new byte[val.length];
-            for (int i = 0; i < val.length; i++) {
-                if (val[i] < 0) {
-                    dst[i] = '?';
-                } else {
-                    dst[i] = val[i];
-                }
-            }
-            return dst;
-        }
-        int len = val.length >> 1;
-        byte[] dst = new byte[len];
-        int dp = 0;
-        for (int i = 0; i < len; i++) {
-            char c = StringUTF16.getChar(val, i);
-            if (c < 0x80) {
-                dst[dp++] = (byte)c;
-                continue;
-            }
-            if (Character.isHighSurrogate(c) && i + 1 < len &&
-                Character.isLowSurrogate(StringUTF16.getChar(val, i + 1))) {
-                i++;
+    @IntrinsicCandidate
+    public static boolean hasNegatives(byte[] ba, int off, int len) {
+        for (int i = off; i < off + len; i++) {
+            if (ba[i] < 0) {
+                return true;
             }
-            dst[dp++] = '?';
-        }
-        if (len == dp) {
-            return dst;
         }
-        return Arrays.copyOf(dst, dp);
-    }
-
-    ////////////////////////// latin1/8859_1 ///////////////////////////
-
-    private static Result decodeLatin1(byte[] ba, int off, int len) {
-       Result result = resultCached();
-       if (COMPACT_STRINGS) {
-           return result.with(Arrays.copyOfRange(ba, off, off + len), LATIN1);
-       } else {
-           return result.with(StringLatin1.inflate(ba, off, len), UTF16);
-       }
+        return false;
     }
 
     @IntrinsicCandidate
-    private static int implEncodeISOArray(byte[] sa, int sp,
+    public static int implEncodeISOArray(byte[] sa, int sp,
                                           byte[] da, int dp, int len) {
         int i = 0;
```

Comment:
As a separate cleanup... 
If there isn't any value to these intrinsics being in StringCoder, they could also move to String
with the corresponding intrinsic references.